plugins/net_plugin/net_plugin.cpp

```diff
@@ -1081,6 +1096,25 @@ namespace eosio {
       });
    }
 
+   void connection::check_heartbeat( const tstamp &current_time )
+   {
+      if( protocol_version >= heartbeat_interval ) {
+         if( latest_msg_time > 0 &&  current_time > latest_msg_time + heartbeat_timeout ) {
+            if( !peer_addr.empty() ) {
+               fc_wlog(logger, "heartbeat timed out for peer address ${adr}", ("adr", peer_addr));
+               no_retry = benign_other;
+            } else {
+               fc_wlog(logger, "heartbeat timed out from ${p} ${ag}", ("p", last_handshake_recv.p2p_address)("ag", last_handshake_recv.agent));
+               no_retry = fatal_other;
+            }
+
+            flush_queues();
```

Comment:
`close()` calls `flush_queues()` don't call here.
Also I think this would be a situation where you would want immediate reconnect. Pass `true` to close.